api/controllers/social_links.py

```diff
@@ -0,0 +1,102 @@
+from api import app
+from fastapi import Depends, HTTPException
+from api.database import get_db
+from api.auth import get_current_user
+from sqlalchemy.orm import Session
+from api.models import User, Artist, SocialLink, SocialMedia
+
+from api.schemas import SocialUpdate, SocialCreate
+
+
+@app.post("/social/add")
+def add_payment_methods(
+    social: SocialCreate,
+    db: Session = Depends(get_db),
+    user: User = Depends(get_current_user),
+):
+    try:
+        artist = db.query(Artist).filter_by(user_id=user.id).first()
+        if not artist:
+            raise HTTPException(404, "Artist not found for this user")
+        existing_socials = dict(
+            list(
+                map(
+                    lambda x: (x.social_media.name, x.social_media.id),
+                    artist.social_links,
+                )
+            )
+        )
```

Comment:
Why is this a dict of a list?  Does it end up making a list of k-v pairs like so?
```
[{name1: id1, name2: id2, ...}]
```